proof_ipa.go

```diff
@@ -352,14 +429,14 @@ func TreeFromProof(proof *Proof, rootC *Point) (VerkleNode, error) {
 		// but not for block validation.
 		values := make([][]byte, NodeWidth)
 		for i, k := range proof.Keys {
-			if len(proof.Values[i]) == 0 {
+			if len(proof.PreValues[i]) == 0 {
 				// Skip the nil keys, they are here to prove
 				// an absence.
 				continue
 			}
 
 			if bytes.Equal(k[:31], info[string(p)].stem) {
-				values[k[31]] = proof.Values[i]
+				values[k[31]] = proof.PreValues[i]
 			}
```

Comment:
I still feel this `if` is a bit strange. We're reconstructing the `values` to be assigned in the leaf when we'll soon create the LeafNode.

But see what we do in `CreatePath(...)`:
- We use that `value` slice [as expected](https://github.com/gballet/go-verkle/blob/master/tree.go#L442-L447).
- We do [some extra assignments from `stemInfo`](https://github.com/gballet/go-verkle/blob/master/tree.go#L462-L464).

Is the latter needed? Or a noop? Or maybe I'm missing something.

Reply:
yeah I can't remember why that is, and removing it seems to work. I won't remove it in this PR but I'm making a note of it in #395